#!/usr/bin/env python
PACKAGE = "dino_nav"

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

gen.add("speed",    int_t,    0, "Speed of the car", 10,  -100, 100)
gen.add("inflation",int_t,    0, "radious of cells inflation", 2,  0, 10)
gen.add("stop_cost",int_t,    0, "cost distance to start steer", 15,  1, 50)
gen.add("grid_dim", int_t,    0, "grid dimension", 100,  25, 1000)
gen.add("zoom",     double_t, 0, "float zoom", 3,  1, 10)
gen.add("enable",   bool_t,   0, "enable the commands",  True)

''''
gen.add("double_param", double_t, 0, "A double parameter",    .5, 0,   1)
gen.add("str_param",    str_t,    0, "A string parameter",  "Hello World")
gen.add("bool_param",   bool_t,   0, "A Boolean parameter",  True)

size_enum = gen.enum([ gen.const("Small",      int_t, 0, "A small constant"),
                       gen.const("Medium",     int_t, 1, "A medium constant"),
                       gen.const("Large",      int_t, 2, "A large constant"),
                       gen.const("ExtraLarge", int_t, 3, "An extra large constant")],
                     "An enum to set size")

gen.add("size", int_t, 0, "A size parameter which is edited via an enum", 1, 0, 3, edit_method=size_enum)
'''

exit(gen.generate(PACKAGE, "dynamic_dinonav", "Dinonav"))
